backend/fms_core/serializers.py

```diff
@@ -608,6 +609,17 @@ def get_released_status_count(self, obj):
     
     def get_blocked_status_count(self, obj):
         return Readset.objects.filter(dataset=obj.id, release_status=ReleaseStatus.BLOCKED).count()
+
+    def get_validation_status(self, obj):
+        counts = Dataset.objects.filter(pk=obj.id) \
```

Comment:
This is clearly not the most effective way to do this ...

Reply:
I would move this logic to the model (as a property also in case we have more than one place that requires (maybe in services) this same information.